packages/react/src/components/Forms/DatePicker/index.tsx

```diff
@@ -158,6 +158,22 @@ export const DatePicker = ({
               )}
             </div>
           </LabelledBox>
+          <Button
+            className={classNames("c__date-picker__inner__action", {
+              "c__date-picker__inner__action--empty": !state.value,
+              "c__date-picker__inner__action--hidden":
+                labelAsPlaceholder || disabled,
+            })}
+            color="tertiary"
+            size="small"
+            icon={<span className="material-icons">cancel</span>}
+            // Intentionally pass a null value to reset date picker state
+            onClick={() => state.setValue(null as unknown as DateValue)}
```

Comment:
Actually, there is an issue open on `react-aria` repo. That's one of the workaround I have found, that was approved by JB on a previous PR. I thought about adding a typescript ignore instead of casting, would you prefer this approach?